### PR TITLE
Toggle Ad Headline Generation with OpenAI

### PR DESCRIPTION
**Pull request by lgaleana-llm**
Add a check to toggle the ad headline generation using OpenAI based on the environment variable USE_OPENAI_HEADLINE. If the variable is set to 'True', the OpenAI API is used to generate the headline. Otherwise, a default text 'Ad headline' is used.

### Test Plan

pytest / playwright

### CODE REVIEW

**Comment by lgaleana-llm** [comment]
### Summary of Test Failures

#### 1. Errors Encountered:
- The test `test_extract_text_success_with_openai` consistently failed with an `AssertionError`.
- The specific error was that the expected headlines generated by OpenAI ("Mocked Ad Headline") were not returned. Instead, the default "Ad headline" was returned.
- The error message indicated a mismatch between the expected and actual headlines:
  ```
  AssertionError: assert {'headlines':.../image2.jpg']} == {'headlines':.../image2.jpg']}
  Differing items:
  {'headlines': ['Ad headline', 'Ad headline']} != {'headlines': ['Mocked Ad Headline', 'Mocked Ad Headline']}
  ```

#### 2. Attempts to Fix the Errors:
- **Initial Fixes:**
  - Ensured that the environment variable `USE_OPENAI_HEADLINE` was correctly set to "True" during the test.
  - Added logging to verify the value of `USE_OPENAI_HEADLINE` within the test and the `extract_text` function.

- **Detailed Logging:**
  - Added detailed logging within the `extract_text` function to trace the flow and verify that the correct branch of the code was executed.
  - Verified that the environment variable was correctly set and logged its value during the test.

- **Mocking Verification:**
  - Ensured that the OpenAI client was correctly mocked and that the mocked client was being used.
  - Added logging to verify the response from the mocked OpenAI client.

- **Simplification:**
  - Simplified the test to isolate the issue and ensure that the OpenAI client was correctly mocked and used.
  - Added additional logging to print the response JSON from the `extract_text` endpoint.

Despite these efforts, the test continued to fail with the same `AssertionError`, indicating that the expected headlines generated by OpenAI were not being returned. The environment variable `USE_OPENAI_HEADLINE` was correctly set to "True", but the logic to generate headlines using OpenAI was not being executed as expected.

**Comment by lgaleana-llm** [comment]
Total time to create PR: 394.03 seconds
Total OPenAI cost: 3.04023

File: app/routes.py [hunk 1]
```diff
@@ -18,6 +18,7 @@
 
 # Initialize OpenAI API key
 openai_api_key = os.getenv("OPENAI_API_KEY")
+use_openai_headline = os.getenv("USE_OPENAI_HEADLINE", "False").lower() == "true"
```
Comment (lgaleana) [inline app/routes.py]: This should check for `== "True"`. This is also being checked at the module level, so make sure that the tests account for that.

Comment (lgaleana-llm) [reply app/routes.py]: Thank you for the feedback, @lgaleana. I'll update the check to `== "True"` and ensure the tests account for the module-level evaluation. I'll push the changes shortly.

**Comment by lgaleana-llm** [comment]
!!!!! WARNING: Tests not in this PR are failing !!!!!

**Comment by lgaleana-llm** [comment]
Total time to amend PR: 625.24 seconds
Total OPenAI cost: 1.99603

File: tests/test_extract_text.py [hunk 2]
```diff
@@ -34,25 +39,56 @@ def mock_requests_get_image(*args, **kwargs):
...
         assert response.status_code == 200
-        assert response.json() == {"images": expected_images, "headlines": expected_headlines}
-        for headline in response.json()["headlines"]:
+        actual_response = response.json()
+        print(f"Actual response: {actual_response}")
+        assert actual_response == {"images": expected_images, "headlines": expected_headlines}
+        for headline in actual_response["headlines"]:
             assert len(headline.split()) <= 5  # Ensure each headline is no more than 5 words
 
+@patch('requests.get', side_effect=mock_requests_get_image)
```
Comment (lgaleana) [inline tests/test_extract_text.py]: Put this in a different file

Comment (lgaleana-llm) [reply tests/test_extract_text.py]: Thank you for the feedback, @lgaleana. I'll move the test to a separate file to keep the tests organized. I'll push the changes shortly.

**Comment by lgaleana-llm** [comment]
Total time to amend PR: 750.92 seconds
Total OPenAI cost: 0.881305

**Comment by lgaleana-llm** [comment]
Total time to amend PR: 765.17 seconds
Total OPenAI cost: 0.0

**Review by lgaleana — APPROVED** [review]

**Pull request merged**